pkg/apis/triggers/v1alpha1/trigger_validation.go

```diff
@@ -70,21 +70,25 @@ func (t TriggerSpecTemplate) validate(ctx context.Context) *apis.FieldError {
 }
 
 func (t triggerSpecBindingArray) validate(ctx context.Context) *apis.FieldError {
-	if len(t) > 0 {
-		for i, b := range t {
-			// Either Ref or Spec should be present
-			if b.Ref == "" && b.Spec == nil {
-				return apis.ErrMissingOneOf(fmt.Sprintf("bindings[%d].Ref", i), fmt.Sprintf("bindings[%d].Spec", i))
-			}
-
-			// Both Ref and Spec can't be present at the same time
-			if b.Ref != "" && b.Spec != nil {
+	for i, b := range t {
+		switch {
+		case b.Ref != "":
+			switch {
+			case b.Spec != nil: // Cannot specify both Ref and Spec
 				return apis.ErrMultipleOneOf(fmt.Sprintf("bindings[%d].Ref", i), fmt.Sprintf("bindings[%d].Spec", i))
-			}
-
-			if b.Ref != "" && b.Kind != NamespacedTriggerBindingKind && b.Kind != ClusterTriggerBindingKind {
+			case b.Name != "": // Cannot specify both Ref and Name
+				return apis.ErrMultipleOneOf(fmt.Sprintf("bindings[%d].Ref", i), fmt.Sprintf("bindings[%d].Name", i))
+			case b.Kind != NamespacedTriggerBindingKind && b.Kind != ClusterTriggerBindingKind: // Kind must be valid
 				return apis.ErrInvalidValue(fmt.Errorf("invalid kind"), fmt.Sprintf("bindings[%d].kind", i))
 			}
+		case b.Spec != nil: // Deprecated old style embedded bindings
```

Comment:
Let's link to an issue and add comments about when this is safe to delete.

Reply:
done